build/tasks/ComposeNewStore.cs

```diff
@@ -0,0 +1,131 @@
+
+        [Required]
+        public ITaskItem[] NewManifests { get; set; }
+
+        [Required]
+        public ITaskItem[] RuntimeStoreFiles { get; set; }
+
+        [Required]
+        public ITaskItem[] RuntimeStoreSymbolFiles { get; set; }
+
+        [Required]
+        public string ManifestDestination { get; set; }
+
+        [Required]
+        public string StoreDestination { get; set; }
+
+        [Required]
+        public string SymbolsDestination { get; set; }
+
+        public override bool Execute()
+        {
+            var existingFiles = new Dictionary<string, HashSet<string>>();
+            var newRuntimeStoreFiles = new List<ITaskItem>();
+            var newRuntimeStoreSymbolFiles = new List<ITaskItem>();
+
+            // Construct database of existing assets
+            foreach (var manifest in ExistingManifests)
+            {
+                var xmlDoc = new XmlDocument();
+                xmlDoc.Load(manifest.GetMetadata("FullPath"));
+                var storeArtifacts = xmlDoc.SelectSingleNode("/StoreArtifacts");
+
+                foreach (XmlNode artifact in storeArtifacts.ChildNodes)
+                {
+                    if (existingFiles.TryGetValue(artifact.Attributes["Id"].Value, out var versions))
+                    {
+                        versions.Add(artifact.Attributes["Version"].Value);
+                    }
+                    else
+                    {
+                        existingFiles[artifact.Attributes["Id"].Value] = new HashSet<string>{ artifact.Attributes["Version"].Value };
+                    }
+                }
+            }
+
+            // Insert new runtime store files
+            foreach (var storeFile in RuntimeStoreFiles)
+            {
+                // format: {bitness}}/{tfm}}/{id}/{version}}/...
+                var recursiveDir = storeFile.GetMetadata("RecursiveDir");
+                var components = recursiveDir.Split(Path.DirectorySeparatorChar);
+                var id = components[2];
+                var version = components[3];
+
+                if (!existingFiles.TryGetValue(id, out var versions) || !versions.Contains(version))
+                {
+                    var destinationDir = Path.Combine(StoreDestination, recursiveDir);
+                    if (!Directory.Exists(Path.Combine(StoreDestination, recursiveDir)))
+                    {
+                        Directory.CreateDirectory(destinationDir);
+                    }
+
+                    File.Copy(storeFile.GetMetadata("FullPath"), Path.Combine(destinationDir, $"{storeFile.GetMetadata("Filename")}{storeFile.GetMetadata("Extension")}"), overwrite: true);
+                }
+            }
+
+            // Insert new runtime store files
+            foreach (var symbolFile in RuntimeStoreSymbolFiles)
+            {
+                // format: {bitness}}/{tfm}}/{id}/{version}}/...
+                var recursiveDir = symbolFile.GetMetadata("RecursiveDir");
```

Comment:
Any time you use file paths coming from MSBuild items, normalize dir separators first to '/'. We've had lots of bugs in the past on Linux from missing this step.

Reply:
What do you mean normalize? https://github.com/dotnet/dotnet-cli-archiver/pull/3/files#diff-e872d285abbe85ecc125a21cdbee8021R7? I'm not doing anything that assumes slash or back-slash since I'm just splitting by DirectorySeparatorChar. Does that not work?